src/Microsoft.AspNetCore.Razor.Language/Extensions/DefaultTagHelperTargetExtension.cs

```diff
@@ -637,6 +638,16 @@ private string GetContent(HtmlContentIntermediateNode node)
             return builder.ToString();
         }
 
+        // Internal for testing
+        internal static string GetDeterministicId(CodeRenderingContext context)
+        {
+            // Use the file checksum along with the absolute position in the generated code to create a unique id for each tag helper call site.
+            var checksum = Checksum.BytesToString(context.SourceDocument.GetChecksum());
```

Comment:
Should this be cached as a property?

Reply:
It's already cached in SourceDocument.

Reply:
If you have a cache you have to worry about invalidating the cache. Would that be a problem here?

Reply:
I don't want to cache anything at this level. The checksum is always 20 bytes and converting that to string shouldn't impact performance.